pkg/segment/reader/metrics/tagstree/tagstreereader.go

```diff
@@ -198,7 +213,11 @@ func (attr *AllTagTreeReaders) FindTSIDS(mQuery *structs.MetricsQuery) (*tsidtra
 							err = tracker.BulkAddStarTagsOnly(rawTagValueToTSIDs, initMetricName, tf.TagKey, numValueFiltersNonZero)
 						} else {
 							initMetricName = fmt.Sprintf("%v{", mQuery.MetricName)
-							err = tracker.BulkAddStar(rawTagValueToTSIDs, initMetricName, tf.TagKey, numValueFiltersNonZero)
+							if tf.TagOperator == segutils.Regex {
```

Comment:
what about `tf.TagOperator == segutils.NegRegex` ? Are we not worried about that case?